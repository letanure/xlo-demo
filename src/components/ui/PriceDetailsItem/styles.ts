import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${() => css`
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    margin-bottom: 24px;
  `}
`

export const Label = styled.div`
  ${({ theme }) => css`
    display: flex;
    font-family: ${theme.font.family.body.name};
    font-size: ${theme.font.size.xsmall};
    font-weight: ${theme.font.weight.normal};
    line-height: ${theme.font.lineHeight.large};
    text-align: left;
    width: 50%;
  `}
`
export const LabelIcon = styled.div`
  ${({ theme }) => css`
    margin-right: ${theme.spacing.xxxsmall};
  `}
`

export const LabelText = styled.div`
  ${() => css``}
`

export const Price = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.font.family.heading.name};
    font-size: ${theme.font.size.small};
    font-weight: ${theme.font.weight.bold1};
    line-height: ${theme.font.lineHeight.large};
    text-align: left;
    width: 50%;
  `}
`

export const PriceRegular = styled.span`
  ${() => css``}
`

export const PriceDiscount = styled.span`
  ${({ theme }) => css`
    text-decoration: line-through;
    color: ${theme.color.primary.disabled};
    color: #7f9799;
    display: inline-block;
    margin-left: 5px;
  `}
`

export const Description = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.font.family.body.name};
    font-size: ${theme.font.size.xsmall};
    font-weight: ${theme.font.weight.normal};
    line-height: ${theme.font.lineHeight.xsmall};
    margin-top: ${theme.spacing.xxxsmall};
    text-align: left;
    width: 100%;
  `}
`
