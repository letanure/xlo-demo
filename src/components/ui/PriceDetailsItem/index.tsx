import * as S from './styles'
import Price, { PriceProps as PriceType } from '../Price'
import Icon, { IconProps } from '../Icon'

export type PriceDetailsItemProps = {
  description?: string
  discount?: PriceType
  icon?: IconProps['name']
  label: string
  price?: PriceType
}

const PriceDetailsItem = ({
  description,
  discount,
  icon,
  label,
  price
}: PriceDetailsItemProps) => (
  <S.Wrapper>
    <S.Label>
      {icon && (
        <S.LabelIcon>
          <Icon name={icon} />
        </S.LabelIcon>
      )}
      <S.LabelText>{label}</S.LabelText>
    </S.Label>
    <S.Price>
      {price && (
        <S.PriceRegular>
          <Price {...price} />
        </S.PriceRegular>
      )}
      {discount && (
        <S.PriceDiscount>
          {' '}
          <Price {...discount} />
        </S.PriceDiscount>
      )}
    </S.Price>
    {description && <S.Description>{description}</S.Description>}
  </S.Wrapper>
)

export default PriceDetailsItem
