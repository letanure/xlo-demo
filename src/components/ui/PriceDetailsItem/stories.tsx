import { ComponentStory, ComponentMeta } from '@storybook/react'

import PriceDetailsItem from '.'

export default {
  title: 'Layout/PriceDetailsItem',
  component: PriceDetailsItem,
  argTypes: {
    label: { control: 'text' },
    price: { control: 'object' },
    description: { control: 'text' },
    discount: { control: 'object' }
  }
} as ComponentMeta<typeof PriceDetailsItem>

const Template: ComponentStory<typeof PriceDetailsItem> = (args) => (
  <PriceDetailsItem {...args} />
)

export const Basic = Template.bind({})
Basic.args = {
  label: 'Label',
  price: {
    amount: 10000,
    currency: 'PLN'
  }
}

export const Icon = Template.bind({})
Icon.args = {
  label: 'Label',
  price: {
    amount: 10000,
    currency: 'PLN'
  },
  icon: 'info'
}

export const Description = Template.bind({})
Description.args = {
  label: 'Label',
  price: {
    amount: 10000,
    currency: 'PLN'
  },
  description:
    'Reference site about Lorem Ipsum, giving information on its origins, as well as a random Lipsum generator.'
}

export const Discount = Template.bind({})
Discount.args = {
  label: 'Label',
  price: {
    amount: 10000,
    currency: 'PLN'
  },
  discount: {
    amount: 15000,
    currency: 'PLN'
  }
}

export const All = Template.bind({})
All.args = {
  label: 'Label',
  price: {
    amount: 10000,
    currency: 'PLN'
  },
  discount: {
    amount: 15000,
    currency: 'PLN'
  },
  description:
    'Reference site about Lorem Ipsum, giving information on its origins, as well as a random Lipsum generator.'
}
