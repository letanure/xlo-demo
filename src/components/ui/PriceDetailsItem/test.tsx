import { render, screen } from '@testing-library/react'

import PriceDetailsItem from '.'

describe('<PriceDetailsItem />', () => {
  it.skip('should render something', () => {
    const { container } = render(<PriceDetailsItem />)

    expect(
      screen.getByRole('heading', { name: /PriceDetailsItem/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
