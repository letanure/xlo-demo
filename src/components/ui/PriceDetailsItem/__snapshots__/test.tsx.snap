// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriceDetailsItem /> should render something 1`] = `
.c0 {
  content: 'default';
}

<div
  class="c0"
>
  <h1>
    PriceDetailsItem with 
    default
  </h1>
</div>
`
